src/App.js

```diff
@@ -319,9 +302,28 @@ class App extends Component {
   );
 }
 
-const AppWithRedux = connect(state => ({
+const OldAppWithRedux = connect(state => ({
   walletReady: state.appState.walletReady,
-}))(App);
+}))(OldApp);
+
+function App() {
+  useAppState({
+    onChange(state) {
+      analytics.track('State change', {
+        category: 'app state',
+        label: state,
+      });
+    },
+    onActive() {
+      store.dispatch(walletConnectLoadState());
+      InteractionManager.runAfterInteractions(() => {
+        rainbowTokenList.update();
+      });
+    },
+  });
+
+  return <OldAppWithRedux />;
```

Comment:
@estrattonbailey Whats the usecase for having an `OldAppWithRedux` would there be a world where this would just be one component eventually?